onyo/lib/ui.py

```diff
@@ -0,0 +1,208 @@
+        Initialize the User Interface object for user communication of Onyo.
+
+        Parameters
+        ----------
+        debug: boolean
+            Activate the debug mode to display additional information via Onyo,
+            and to print the full traceback stack if errors occur.
+
+        quiet: boolean
+            Activate the quiet mode (requires that `yes=True`) to suppress all
+            output generation.
+
+        yes: boolean
+            Activate the yes mode to suppress all interactive requests to the
+            user, and instead answers them with yes.
+        """
+        # set the the attributes of the UI object
+        self.quiet = quiet
+        self.yes = yes
+        self.logger = logging.getLogger('onyo')
+
+        # set the debug level
+        if debug:
+            self.logger.setLevel(logging.DEBUG)
+        else:
+            self.logger.setLevel(logging.INFO)
+
+    def set_debug(self,
+                  debug: bool = False) -> None:
+        """
+        Set the log level to activate debug mode.
+
+        Parameters
+        ----------
+        debug: boolean
+            Activates debug mode, and configures the log level of the logger.
+        """
+        if debug:
+            self.logger.setLevel(logging.DEBUG)
+        else:
+            self.logger.setLevel(logging.INFO)
+
+    def set_quiet(self,
+                  quiet: bool = False) -> None:
+        """
+        Set `quiet` to suppress terminal output.
+
+        Parameters
+        ----------
+        quiet: boolean
+            `True` suppresses of all user output.
+            Requires `yes` mode to be active.
+
+        Raises
+        ------
+        ValueError
+            If tried to activate quiet mode without `yes=True`.
+        """
+        if quiet and not self.yes:
+            raise ValueError("The --quiet flag requires --yes.")
+        self.quiet = quiet
+
+    def set_yes(self,
+                yes: bool = False) -> None:
+        """
+        Set `yes` to answer all requests to user with "yes".
+
+        Parameters
+        ----------
+        yes: boolean
+            Activate yes mode, which suppresses all user requests and answers
+            them positively. Allows the activation of the quiet mode.
+        """
+        self.yes = yes
+
+    def error(self,
+              error: Union[str, Exception],
+              end: str = "\n") -> None:
+        """
+        Print an error message, if the UI is not set to `quiet`.
+
+        Parameters
+        ----------
+        error: string or Exception
+            Prints the string, or the message of an error.
+            If debug mode is activated, displays the full traceback of an error.
+
+        end: str
+            Specify the string at the end of prints.
+            Per default, prints end with a line break.
+        """
+        if not self.quiet:
+            print(f"ERROR: {error}", file=sys.stderr, end=end)
+        self.logger.debug(format_exc())
```

Comment:
I'd use https://docs.python.org/3/library/traceback.html#traceback.format_exception and actually pass `error` (if it is an `Exception`. Otherwise you are relying on being in a context where an exception was just caught. We don't really know that here. Caller gave the exception object to consider explicitly. Let's use it.